modules/core/src/smithy4s/internals/DocumentDecoderSchemaVisitor.scala

```diff
@@ -164,6 +165,22 @@ class DocumentDecoderSchemaVisitor(
       from("Byte") {
         case FlexibleNumber(bd) if bd.isValidByte => bd.toByte
       }
+    case PLocalDate =>
+      fromUnsafe("LocalDate") { case DString(string) =>
+        LocalDate.parseUnsafe(string)
+      }
+    case PLocalTime =>
+      from("LocalTime") { case DString(string) =>
+        LocalTime.parseUnsafe(string)
+      }
+    case PDuration =>
+      from("Duration") { case DString(string) =>
+        Duration(string)
+      }
+    case POffsetDateTime => 
+      from("OffsetDateTime") { case DString(string) =>
+        OffsetDateTime.parseUnsafe(string)
```

Comment:
I think these should use `fromUnsafe` instead of `from` so that `PayloadError` will be thrown ultimately instead of random errors